File: java/src/patterns/creational/factory/concretecreators/JsonParserCreator.java
package patterns.creational.factory.concretecreators;

import patterns.creational.factory.abstractcreator.ParserCreator;
import patterns.creational.factory.concreteproducts.JsonParser;
import patterns.creational.factory.productinterface.DocumentParser;

public class JsonParserCreator extends ParserCreator {
    @Override
    public DocumentParser createParser() {
        return new JsonParser();
    }
}

File: python/src/patterns/creational/factory/concrete_creators/json_parser_creator.py
from src.patterns.creational.factory.abstract_creator.parser_creator import ParserCreator
from src.patterns.creational.factory.concrete_products.json_parser import JsonParser
from src.patterns.creational.factory.product_interface.document_parser import DocumentParser

class JsonParserCreator(ParserCreator):
    def create_parser(self) -> DocumentParser:
        return JsonParser()
